test/functional/sqlsrv/TC47_ParamQuery.phpt

```diff
@@ -11,27 +11,34 @@ PHPT_EXEC=true
 <?php
 require_once('MsCommon.inc');
 
-function ParamQuery($minType, $maxType)
+function paramQuery($minType, $maxType)
 {
-    include 'MsSetup.inc';
-
     $testName = "Parameterized Query";
     startTest($testName);
 
     setup();
-    $conn1 = connect();
+    $tableName = 'TC47test';
+    $conn1 = AE\connect();
 
     for ($k = $minType; $k <= $maxType; $k++) {
-        $data = GetSampleData($k);
+        $data = getSampleData($k);
         if ($data != null) {
-            $sqlType = GetSqlType($k);
-            $phpDriverType = GetDriverType($k, strlen($data));
-            $dataType = "[c1] int, [c2] $sqlType";
-            $dataOptions = array($k, array($data, SQLSRV_PARAM_IN, SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR), $phpDriverType));
+            $sqlType = getSqlType($k);
+            $phpDriverType = getDriverType($k, strlen($data));
+            $columns = array(new AE\ColumnMeta('int', 'c1'),
+                             new AE\ColumnMeta($sqlType, 'c2'));
+            traceData($sqlType, $data);
+
+            $res = null;
+            AE\createTable($conn1, $tableName, $columns);
+            AE\insertRow($conn1,
+                $tableName,
+                array("c1"=>$k, "c2"=>array($data, SQLSRV_PARAM_IN, SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR), $phpDriverType)), 
+                $res,
```

Comment:
can you just put null here?

Reply:
I just tried to be faithful to the original test, in which it called a function insertRowEx() that does the following:
``` $stmt = sqlsrv_query($conn, "INSERT INTO [$tableName] ($dataCols) VALUES ($dataValues)", $dataOptions);```

Reply:
?? I was talking about $res

Reply:
nope... I will get the error `Fatal error: Uncaught Error: Cannot pass parameter 4 by reference `